scripts/build.sh

```diff
@@ -8,13 +8,30 @@ set -o pipefail
 export TARGET=${TARGET:-Release}
 export COVERAGE=${COVERAGE:-false}
 export NODE=${NODE:-4}
+export platform=$(echo $(uname -s)| sed "y/ABCDEFGHIJKLMNOPQRSTUVWXYZ/abcdefghijklmnopqrstuvwxyz/")
+if [[ ${platform} == 'darwin' ]]; then
+    export platform=osx;
+fi
+export CLANG_VERSION="${CLANG_VERSION:-3.8.1}"
+export CMAKE_VERSION="${CMAKE_VERSION:-3.5.2}"
 
 export CURRENT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 
-source ${CURRENT_DIR}/travis_helper.sh
+export DEPS_DIR="$(pwd)/deps"
+export PATH=${DEPS_DIR}/bin:${PATH}
+mkdir -p ${DEPS_DIR}
+
+# install build deps from s3 (without mason client to keep things simple)
+# install cmake
+wget --quiet -O - https://mason-binaries.s3.amazonaws.com/${platform}-x86_64/cmake/${CMAKE_VERSION}.tar.gz | tar --strip-components=1 -xz -C ${DEPS_DIR} || exit 1
```

Comment:
Realizing the catch is that we need to set up the compiler in a way that both osrm-backend and node-osrm can get access to it. The current solution installs the compiler first, but before `osrm-backend` has been download, so we don't yet have access to the vendored mason command. Hrm...